site/pages/api/add-user.api.ts

```diff
@@ -0,0 +1,67 @@
+import { UsernameExistsException } from "@aws-sdk/client-cognito-identity-provider";
+import { NextApiRequest, NextApiResponse } from "next";
+import { ADD_USER_PAGE_PATH, COOKIES_ADD_USER_ERRORS, USER_MANAGEMENT_PAGE_PATH } from "../../constants";
+import { createUser } from "../../data/cognito";
+import { addUserSchema } from "../../schemas/add-user.schema";
+import { flattenZodErrors } from "../../utils";
+import {
+    redirectToError,
+    setCookieOnResponseObject,
+    redirectTo,
+    destroyCookieOnResponseObject,
+} from "../../utils/apiUtils";
+
+const addUser = async (req: NextApiRequest, res: NextApiResponse) => {
+    try {
+        const validatedBody = addUserSchema.safeParse(req.body);
+        if (!validatedBody.success) {
+            setCookieOnResponseObject(
+                COOKIES_ADD_USER_ERRORS,
+                JSON.stringify({
+                    inputs: req.body as object,
+                    errors: flattenZodErrors(validatedBody.error),
+                }),
+                res,
+            );
+
+            redirectTo(res, ADD_USER_PAGE_PATH);
+            return;
+        }
+
+        await createUser(validatedBody.data);
```

Comment:
We should get the session of the logged in user and make sure that the user they're trying to create is in the same org, otherwise admin users could create other users in any organisation

Reply:
done. There is no input for org id. The id is retrieved from the session within the api and passed to cognito call.